test/truffle/compiler/pe/core/string_pe.rb

```diff
@@ -12,4 +12,24 @@
 example "Truffle::Primitive.create_simple_string.getbyte(0)", simple_string.getbyte(0)
 
 example "'abc'.length", 3
+example "'こにちわ'.length", 4
+
+example "'abc'.bytesize", 3
+example "'こにちわ'.bytesize", 12
+
 tagged_example "'abc' == 'abc'", true # seems to fail sometimes
+example "x = 'abc'; x == x", true
+example "x = 'abc'; x == x.dup", true
```

Comment:
This is where the optimisation magic starts! This dup has been completely eliminated by the PE.